File: field-play-frontend/src/interfaces/session/SessionReadDto.ts
import {DateTime} from "luxon";
import {Status} from "./Status.ts";

export default interface SessionReadDto {
    id: number;
    fieldId: number;
    fieldName: string;
    bookingPrice: number;
    signUpPrice: number;
    minPlayers: number;
    maxPlayers: number;
    signUpCount: number;
    status: Status;
    startsAt: DateTime;
}

File: field-play-backend/src/main/java/com/rehneo/fieldplaybackend/session/SessionReadDto.java
package com.rehneo.fieldplaybackend.session;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SessionReadDto {
    private int id;
    private int fieldId;
    private String fieldName;
    private int bookingPrice;
    private int signUpPrice;
    private int minPlayers;
    private int maxPlayers;
    private int signUpCount;
    private Status status;
    private ZonedDateTime startsAt;
    private boolean signedUp;
    private boolean booked;
}
